node/resolvers/address.ts

```diff
@@ -36,45 +35,35 @@ function getCountry(place: Place) {
     : null
 }
 
-function getLanguage(vtex: IOContext) {
-  const language = vtex.locale?.replace('_', '-') ?? ''
-
-  if (!(language in Language)) {
-    vtex.logger.warn(
-      `"${language}" is not a valid language. See the list of supported languages on https://developers.google.com/maps/faq#languagesupport`
-    )
-  }
-
-  return language as Language
-}
-
 const getAddress = async (
   _: unknown,
-  args: QueryAddressArgs,
-  ctx: Context
+  { externalId, sessionToken }: QueryAddressArgs,
+  { clients: { apps, google }, vtex: { logger, locale } }: Context
 ): Promise<Address> => {
-  const { clients, vtex } = ctx
-  const { externalId, sessionToken } = args
-  const { apiKey } = await clients.apps.getAppSettings(process.env.VTEX_APP_ID)
-
-  const client = clients.google
+  const { apiKey } = await apps.getAppSettings(process.env.VTEX_APP_ID)
 
   if (!sessionToken) {
-    vtex.logger.warn('No session token found. Additional charges may apply')
+    logger.warn('No session token found. Additional charges may apply')
+  }
+
+  if (!Object.values(Language).includes(locale as Language)) {
+    logger.warn(
+      `"${locale}" is not a valid language. See the list of supported languages on https://developers.google.com/maps/faq#languagesupport`
+    )
   }
 
-  const response = await client.placeDetails({
+  const response = await google.placeDetails({
     params: {
       place_id: externalId,
-      language: getLanguage(vtex),
+      language: locale ? (locale as Language) : undefined,
```

Comment:
Quite inconsistent with https://github.com/vtex-apps/google-geolocation-resolver/pull/7/files#diff-5661c3a29ad7cb5851273e4604b1a14fbd5cfdf3e6590c9e9bdc33589b0771dbR31, but that's a problem with the lib :/ (as this is a breaking change, I don't know if it's worth it to submit a PR for them)